### PR TITLE
Update `BTSepaDirectDebitRequest`

### PR DESCRIPTION
**Pull request by agedd**
### Summary of changes

- Update `BTSepaDirectDebitRequest` properties to have everything in the init vs using dot syntax

### Checklist

- [X] Added a changelog entry

- @agedd

### CODE REVIEW

File: Sources/BraintreeSEPADirectDebit/BTSEPADirectDebitRequest.swift [hunk 1]
```diff
@@ -7,28 +7,15 @@ import BraintreeCore
...
-    public var billingAddress: BTPostalAddress?
-
-    /// Optional. A non-default merchant account to use for tokenization.
-    public var merchantAccountID: String?
-
-    /// Optional. A locale code to use for creating a mandate.
-    /// See https://developer.paypal.com/reference/locale-codes/ for a list of possible values.
-    /// Locale code should be supplied as a BCP-47 formatted locale code.
-    public var locale: String?
+    var accountHolderName: String?
```
Comment (jaxdesmarais) [inline Sources/BraintreeSEPADirectDebit/BTSEPADirectDebitRequest.swift:12]: Lets remove the optionality from anything marked as required both here and in the init.

**Review by warmkesselj — APPROVED** [review]
🍏

**Review by richherrera — APPROVED** [review]
💯

File: Sources/BraintreeSEPADirectDebit/BTSEPADirectDebitRequest.swift [hunk 2]
```diff
@@ -42,11 +29,11 @@ import BraintreeCore
...
     public init(
-        accountHolderName: String? = nil,
-        iban: String? = nil,
-        customerID: String? = nil,
+        accountHolderName: String,
+        iban: String,
+        customerID: String,
         mandateType: BTSEPADirectDebitMandateType? = .oneOff,
-        billingAddress: BTPostalAddress? = nil,
+        billingAddress: BTPostalAddress,
```
[the remark this answers is not in the document]
Comment (agedd) [reply Sources/BraintreeSEPADirectDebit/BTSEPADirectDebitRequest.swift:35]: yep, addressed in 0ba2902df8f9163b92c0d33652e9072160430f19

File: Sources/BraintreeSEPADirectDebit/BTSEPADirectDebitRequest.swift [hunk 3]
```diff
@@ -7,28 +7,15 @@ import BraintreeCore
...
-    public var billingAddress: BTPostalAddress?
-
-    /// Optional. A non-default merchant account to use for tokenization.
-    public var merchantAccountID: String?
-
-    /// Optional. A locale code to use for creating a mandate.
-    /// See https://developer.paypal.com/reference/locale-codes/ for a list of possible values.
-    /// Locale code should be supplied as a BCP-47 formatted locale code.
-    public var locale: String?
+    var accountHolderName: String
```
Comment (richherrera) [inline Sources/BraintreeSEPADirectDebit/BTSEPADirectDebitRequest.swift:12]: I was thinking, do these optional properties need to be `var`, or can we change them to `let`?

Comment (agedd) [reply Sources/BraintreeSEPADirectDebit/BTSEPADirectDebitRequest.swift:12]: hmm that's a good question! i think it's safe to change to `let` but android also has them defined as `var` so thinking to keep as is to maintain consistency

**Comment by jaxdesmarais** [comment]
Are we able to update the changelog and migration guide language to match what is in https://github.com/braintree/braintree_ios/pull/1436 once that is merged in?

**Comment by agedd** [comment]
> Are we able to update the changelog and migration guide language to match what is in #1436 once that is merged in?

yep, addressed it in e07d2b4d97d78c12525d86319143705d8432eadc!

**Pull request merged**